pokemongo_bot/cell_workers/sniper.py

```diff
@@ -476,20 +499,25 @@ def _hash(self, pokemon):
     def _equals(self, pokemon_1, pokemon_2):
         return self._hash(pokemon_1) == self._hash(pokemon_2)
 
-    def _is_cached(self, pokemon):
-        for cached_pokemon in self.cache:
-            if self._equals(pokemon, cached_pokemon):
-                return True
-
+    def _is_cached(self, uniqueid):
+        if uniqueid in self.bot.sniper_cache:
+            return True
         return False
 
-    def _cache(self, pokemon):
-        # Skip repeated items
-        if not self._is_cached(pokemon):
+    def _cache(self, uniqueid):
+        if not self._is_cached(uniqueid):
             # Free space if full and store it
-            if len(self.cache) >= self.MAX_CACHE_LIST_SIZE:
-                self.cache.pop(0)
-            self.cache.append(pokemon)
+            if len(self.bot.sniper_cache) >= self.MAX_CACHE_LIST_SIZE:
+                self.bot.sniper_cache.pop(0)
+            self.bot.sniper_cache.append(uniqueid)
+
+    def _build_unique_id(self, pokemon):
```

Comment:
Unless you're talking about the len(self.bot.sniper_cache) thing... which is your code so you tell me why it's necessary.